### DISCUSSION TITLE
How can I reduce the size of my `target` folder?

### DISCUSSION THREAD

**Discussion by AribYadi** [bevyengine/bevy#9248] [Q&A]
I'm using [bevy_game_template](https://github.com/NiklasEi/bevy_game_template) and just building it alone gives me a 4gb target folder. For my old computer, that is too much. Is there anything I can do to minimize the size?

**Comment by nicopap** [comment]
cargo keeps around old build artifacts. You'll accumulate over time useless generated files in the target directory. So use `cargo clean`. The next build will be done from scratch and only generate the necessary files for the build. Subsequent builds will not add much more bloat to your target directory.

When does an old build artifact become useless?

1. You are using a new rustc version. Artifacts built with the old rustc version will co-exist with the artifacts of the new version, doubling your target directory size
2. You update dependencies or change their feature set.
3. Release/debug builds also have their own separate artifacts

I'd say a reasonable target size is 1GB.

`cargo clean` implies you'll have to build everything from scratch, and on an old computer this can take several minutes, if not hours. So this isn't a great solution.

But there isn't much else to do. Unless you allow for some byzantine config like a networked drive/distributed builds.

**Comment by wazawoo** [comment]
My `target/debug/deps` folder is 25 GB!!! I'm probably doing something stupid, but that's just insane

**Reply by 17shka** [reply]
in that case, I'm a total idiot with 60 GB

**Comment by dothanhtrung** [comment]
What I often do are:

1. Disable `default-features` in most of my dependencies, and only enable what are required for my project.
2. Periodically clean old data in `target` folder with `cargo sweep` and cronjob. For example:
    ```
    12 11 * * * cd /path/to/my/project && cargo sweep -t 15
    ```
